app/models/concerns/orchestration/dhcp.rb

```diff
@@ -82,25 +82,15 @@ def boot_server
     # if we don't manage tftp for IPv4 at all, we dont create a next-server entry.
     return unless tftp?
 
-    # first try to ask our IPv4 TFTP server for its boot server
-    bs = tftp.bootServer
-    # if that failed, trying to guess out tftp next server based on the smart proxy hostname
-    bs ||= URI.parse(subnet.tftp.url).host
+    unless subnet.dhcp.has_capability?(:DHCP, :dhcp_filename_hostname)
+      failure _("Unable to determine boot server. The DHCP smart proxy doesn't provide dhcp_filename capability or failed to provide this information.")
+    end
 
     # only smart proxies with V2 API are supported
-    if subnet.dhcp.has_capability?(:DHCP, :dhcp_filename_hostname)
-      # we no longer convert the boot server to IPv4 - smart proxy modules are required to do so if they don't support hostname
-      bs
-    else
-      begin
-        Foreman::Deprecation.deprecation_warning('1.25', "DHCP proxy does not report dhcp_filename_* capability and reverse DNS search in Foreman will be removed.")
-        ip = NicIpResolver.new(:nic => self).to_ip_address(bs) if bs.present?
-        return ip unless ip.nil?
-        failure _("Unable to determine the host's boot server. The DHCP smart proxy failed to provide this information and this subnet is not provided with TFTP services.")
-      rescue => e
-        failure _("failed to detect boot server: %s") % e, e
-      end
-    end
+    # we no longer convert the boot server to IPv4 - smart proxy modules are required to do so if they don't support hostname
+    # first try to ask our IPv4 TFTP server for its boot server
```

Comment:
Does this comment help? I'm not sure it really clarifies things for anyone reading this to know that in the past we did a (IP ->) DNS -> IP conversion. Today it's just passing an entry to the Smart Proxy and the Smart Proxy is responsible for any conversion if needed. That's why we have the capability check.